### ISSUE TITLE
Terrain gem: Terrain material sometimes is displayed incorrectly 

### ISSUE THREAD

**Issue by pawelbudziszewski** [o3de/o3de#16929]
**Describe the bug**
The terrain material is displayed only on a part of the terrain if `Surface Data Query Resolution` is lower than 0.5.
In the scene, there is a terrain with a default material defined in the `Surface Material List`. There are no other features. It works fine, if `Surface Data Query Resolution` (in `Terrain World`) is set to 0.5 or higher. If it is set to 0.4 or lower, the material is displayed only in part of the terrain. It looks like it is the part with X>0 and Y>0.

Note: To prevent crash, I need to use a workaround to change the `Surface Data Query Resolution` (see #16927)

**Screenshots/Video**
Surface Data Query Resolution = 0.5
![Screenshot from 2023-10-22 23-52-16](https://github.com/o3de/o3de/assets/31925544/9ee15053-1f00-4905-a682-0e32bd8a3222)

Surface Data Query Resolution = 0.4
![Screenshot from 2023-10-22 23-51-52](https://github.com/o3de/o3de/assets/31925544/976a2aee-87ed-4cf3-a0d1-be758bc1019a)

Setup
![Screenshot from 2023-10-22 23-52-31](https://github.com/o3de/o3de/assets/31925544/c08ec26f-6871-4d2b-9f99-69465693e92d)

**Found in Branch**
Main

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
0d490a2b4b5e04fd17edf4850c91f94361d4b836

**Desktop/Device (please complete the following information):**
Ubuntu Linux

**Comment by moudgils** [comment]
@mbalfour-amzn Is this issue something you have run into recently?

**Comment by mbalfour-amzn** [comment]
@moudgils yep, independently saw it with very different repro steps on one of my terrain test levels last week. in my case, i thought maybe it was a vulkan implementation bug involving non-power-of-2 buffers, since i was setting a size of something to 1500x1500 when it failed, and 1024x1024 worked just fine.

**Comment by mbalfour-amzn** [comment]
A couple more notes from investigating further:
1. I believe this is Vulkan-specific, I can't get it to happen on DX12.
2. I believe the core problem is that the staging buffers on Vulkan are defined to be 128 MB by default, and so the terrain system is running out of memory in the staging buffers:
* https://github.com/o3de/o3de/blob/development/Gems/Atom/RHI/Vulkan/Code/Source/RHI/Device.cpp#L43-L45
* https://github.com/o3de/o3de/blob/development/Gems/Atom/RHI/Code/Include/Atom/RHI.Reflect/PlatformLimitsDescriptor.h#L39-L51

The only part that's odd is that while I'm immediately hitting out of memory asserts from the staging pool on Windows, I don't seem to be hitting those asserts on Linux. So there might be a secondary problem going on here as well.

**Comment by mbalfour-amzn** [comment]
[TestTerrain.zip](https://github.com/o3de/o3de/files/13209876/TestTerrain.zip)
Here's a level that demonstrates the problem in the AutomatedTesting project.

The entire terrain _should_ draw as green to a distance of N meters around the camera based on the "Detail Material Render Distance" on the Terrain World Renderer component. With the level saved as 1030 meters, it's only drawing one quadrant correctly. If I set it to 1024 or 1028 or 2048, it draws everything correctly. On DX12, it draws correctly for all values. 

Here it is set to 1030 (incorrect):
![image](https://github.com/o3de/o3de/assets/82224783/b6782ff2-9061-49d4-ade9-5f3939830c75)

Here it is set to 1028 (correct):
![image](https://github.com/o3de/o3de/assets/82224783/7b7c261b-53dd-4ea8-9496-99b5d650cb24)
